performance_test/src/communication_abstractions/ros2_callback_communicator.hpp

```diff
@@ -44,6 +44,7 @@ class ROS2QOSAdapter
   inline rmw_qos_profile_t get() const
   {
     rmw_qos_profile_t ros_qos;
+    memset(&ros_qos, 0, sizeof(ros_qos));
```

Comment:
Could you explain this a little bit?

Reply:
I found that at least the "avoid_ros_namespace_conventions" field in the QoS was undefined, and that caused the CycloneDDS RMW layer to sometimes use one topic name, and sometimes another. It seemed very unlikely that undefined fields was intended behaviour, and memsetting ``ros_qos`` seemed the most sensible way of solving the problem.

Reply:
Interesting, thanks for looking into it. This should probably be addressed upstream in ROS 2, but it's ok to keep it here until then.